homeassistant/components/sensor/forecast.py

```diff
@@ -48,21 +50,9 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Forecast.io sensor."""
-    import forecastio
-
-    if None in (hass.config.latitude, hass.config.longitude):
-        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
-        return False
-
-    try:
-        forecast = forecastio.load_forecast(config.get(CONF_API_KEY, None),
-                                            hass.config.latitude,
-                                            hass.config.longitude)
-        forecast.currently()
-    except ValueError:
-        _LOGGER.error(
-            "Connection error "
-            "Please check your settings for Forecast.io.")
+    api_key = config.get(CONF_API_KEY, None)
+    if None in (hass.config.latitude, hass.config.longitude, api_key):
```

Comment:
I was able to use `validate_config` for the API key, but couldn't figure out how to use it for a HA level config. All the example I found were just checking manually like above